CMakeModules/build_cl2hpp.cmake

```diff
@@ -0,0 +1,28 @@
+INCLUDE(ExternalProject)
+
+SET(prefix ${CMAKE_BINARY_DIR}/third_party/cl2hpp/src/cl2hpp-ext-build)
+
+ExternalProject_Add(
+    cl2hpp-ext
+    GIT_REPOSITORY https://github.com/KhronosGroup/OpenCL-CLHPP.git
+    GIT_TAG 2b415bf8fc6ab035b6de6a14f3c579f91199fa2a
+    PREFIX "${prefix}"
+    INSTALL_COMMAND ""
+    INSTALL_DIR "${prefix}"
+    UPDATE_COMMAND ""
+    CONFIGURE_COMMAND ${CMAKE_COMMAND} -Wno-dev "-G${CMAKE_GENERATOR}" <SOURCE_DIR>
+    -DCMAKE_SOURCE_DIR:PATH=<SOURCE_DIR>
+    -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
+    -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
+    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+    -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+    -DBUILD_DOCS:BOOL=OFF
+    -DBUILD_EXAMPLES:BOOL=ON
+    -DBUILD_TESTS:BOOL=OFF
+    )
```

Comment:
Hmm this might be a problem for people who don't have python on their machines. 

Can we host an auto-generated cl2.hpp someplace and download that instead?